src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/LogHubLogger.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Logging
+{
+    internal class LogHubLogger : ILogger
+    {
+        private readonly string _categoryName;
+        private readonly LogHubLogWriter _logWriter;
+        private readonly Scope _noopScope;
+
+        public LogHubLogger(string categoryName, LogHubLogWriter feedbackFileLogWriter)
+        {
+            if (categoryName is null)
+            {
+                throw new ArgumentNullException(nameof(categoryName));
+            }
+
+            if (feedbackFileLogWriter is null)
+            {
+                throw new ArgumentNullException(nameof(feedbackFileLogWriter));
+            }
+
+            _categoryName = categoryName;
+            _logWriter = feedbackFileLogWriter;
+            _noopScope = new Scope();
+        }
+
+        public IDisposable BeginScope<TState>(TState state) => _noopScope;
+
+        public bool IsEnabled(LogLevel logLevel) => true;
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            var formattedResult = formatter(state, exception);
+            var logContent = $"[{_categoryName}] {formattedResult}";
```

Comment:
Is there a more performant write API on the trace source that we can call instead of re-building a string like this eveyr log call? If we can avoid these allocations using these new APIs that'd be best

Reply:
Updated to leverage the `TraceSource` formatting APIs, however I doubt that's doing anything differently than this :/. We could of course just remove the `categoryName`, however personally I find it helpful when investigating the logs.

Reply:
Category name is definitely helpful. If there's no way around it then we're stuck for now